src/Domain/UseCase/SignIn.spec.ts

```diff
@@ -0,0 +1,53 @@
+import 'reflect-metadata'
+
+import { AuthResponseFactoryInterface } from '../Auth/AuthResponseFactoryInterface'
+import { CurrentAuthResponse } from '../Auth/CurrentAuthResponse'
+import { User } from '../User/User'
+import { UserRepositoryInterface } from '../User/UserRepositoryInterface'
+import { SignIn } from './SignIn'
+
+describe('SignIn', () => {
+  let user: User
+  let authResponse: CurrentAuthResponse
+  let userRepository: UserRepositoryInterface
+  let authResponseFactory: AuthResponseFactoryInterface
+
+  const createUseCase = () => new SignIn(userRepository, authResponseFactory)
+
+  beforeEach(() => {
+    user = {} as jest.Mocked<User>
+    user.encryptedPassword = '$2a$11$K3g6XoTau8VmLJcai1bB0eD9/YvBSBRtBhMprJOaVZ0U3SgasZH3a'
+
+    userRepository = {} as jest.Mocked<UserRepositoryInterface>
+    userRepository.findOneByEmail = jest.fn().mockReturnValue(user)
+
+    authResponse = {} as jest.Mocked<CurrentAuthResponse>
+
+    authResponseFactory = {} as jest.Mocked<AuthResponseFactoryInterface>
+    authResponseFactory.createSuccessAuthResponse = jest.fn().mockReturnValue(authResponse)
+  })
+
+  it('should sign in a user', async () => {
+    expect(await createUseCase().execute({
+      email: 'test@test.te',
+      password: 'qweqwe123123',
+      userAgent: 'Google Chrome',
+      apiVersion: '004'
```

Comment:
apiVersion does not correspond to 004, 003, etc, but 20190520, 20190520, 20161215. Not sure if this is just a dummy value in this test.

Reply:
yeah, it's a dummy version - I'll replace it so it doesn't induce confusion